face_swapping/facefusion/facefusion.py

```diff
@@ -0,0 +1,238 @@
+import cv2
+import sys
+import time
+import onnx
+import numpy as np
+from onnx import numpy_helper
+
+import ailia
+
+# import original modules
+sys.path.append('../../util')
+from arg_utils import get_base_parser, update_parser, get_savepath  # noqa
+from model_utils import check_and_download_models  # noqa
+from webcamera_utils import get_capture, get_writer  # noqa
+
+# logger
+from logging import getLogger  # noqa
+
+from facefusion_utils.face_analyser import get_one_face, get_average_face
+from facefusion_utils.utils import read_static_images, read_static_image, write_image
+from facefusion_utils.face_store import append_reference_face
+from facefusion_utils.face_swapper import pre_process, post_process
+from facefusion_utils.face_swapper import get_reference_frame as fs_get_reference_frame, process_image as fs_process_image
+from facefusion_utils.face_enhancer import get_reference_frame as fe_get_reference_frame, process_image as fe_process_image
+
+logger = getLogger(__name__)
+
+# ======================
+# Parameters
+# ======================
+
+WEIGHT_FACE_DETECTOR_PATH = 'yoloface_8n.onnx'
+MODEL_FACE_DETECTOR_PATH = 'yoloface_8n.onnx.prototxt'
+WEIGHT_FACE_LANDMARKER_PATH = '2dfan4.onnx'
+MODEL_FACE_LANDMARKER_PATH = '2dfan4.onnx.prototxt'
+WEIGHT_FACE_RECOGNIZER_PATH = 'arcface_w600k_r50.onnx'
+MODEL_FACE_RECOGNIZER_PATH = 'arcface_w600k_r50.onnx.prototxt'
+WEIGHT_FACE_SWAPPER_PATH = 'inswapper_128.onnx'
+MODEL_FACE_SWAPPER_PATH = 'inswapper_128.onnx.prototxt'
+WEIGHT_FACE_ENHANCER_PATH = 'gfpgan_1.4.onnx'
+MODEL_FACE_ENHANCER_PATH = 'gfpgan_1.4.onnx.prototxt'
+REMOTE_PATH = 'https://storage.googleapis.com/ailia-models/facefusion/'
+
+TARGET_IMAGE_PATH = 'target.jpg'
+SOURCE_IMAGE_PATH = 'source.jpg'
+SAVE_IMAGE_PATH = 'output.jpg'
+
+FACE_DETECTOR_SCORE = 0.5
+REFERENCE_FACE_DISTANCE = 0.6
+
+# ======================
+# Argument Parser Config
+# ======================
+
+parser = get_base_parser(
+    'Facefusion', TARGET_IMAGE_PATH, SAVE_IMAGE_PATH
+)
+parser.add_argument(
+    '-src', '--source', nargs='+', metavar='IMAGE', default=[SOURCE_IMAGE_PATH,],
+    help=('The source image(s) to swap the face from.')
+)
+parser.add_argument(
+    '--skip_enhance', action='store_true',
+    help='Whether to skip face enhancement using GFPGAN.'
+)
+parser.add_argument(
+    '-th', '--threshold', type=float, default=FACE_DETECTOR_SCORE,
+    help='Face detector score threshold.'
+)
+parser.add_argument(
+    '-dist', '--face_distance', type=float, default=REFERENCE_FACE_DISTANCE,
+    help='Face distance similarity score threshold.'
+)
+parser.add_argument(
+    '--onnx',
+    action='store_true',
+    help='Execute onnxruntime version.'
+)
+args = update_parser(parser)
+
+# ======================
+# Secondary functions
+# ======================
+
+def get_model_matrix(model_path):
```

Comment:
Can you remove dependency for onnx?
Maybe you can use np.save and np.load.

Reply:
Sorry about that, it should be done now.